pandas/tseries/index.py

```diff
@@ -1581,10 +1590,14 @@ def insert(self, loc, item):
         """
 
         freq = None
+
+        if isinstance(item, np.datetime64):
+            # np.datetime64 doesn't have tz, once convert to Timestamp
+            item = Timestamp(item)
```

Comment:
In ``_has_same_tz``, ``other`` can be DTI or list-likes. Thus added ``if`` for ``np.datetime64``.